Kernel crash in a Docker container environment

### Applies To

- [X] Notebooks (.ipynb files)
- [ ] Interactive Window and\/or Cell Scripts (.py files with \#%% markers)

### What happened?

Kernel crash happens when running cells in a .ipynb on docker container environment.
We run the same notebook on the local environment successfully though.


### VS Code Version

1.85.1

### Jupyter Extension Version

v2023.11.1100101639

### Jupyter logs

```shell
01:14:57.278 [info] Handle Execution of Cells 10 for /work/python/sample/sample_mapping_ipynb/sample_mapping_1.ipynb
01:14:57.285 [info] Starting Kernel startUsingPythonInterpreter, .jvsc74a57bd0949777d72b0d2535278d3dc13498b2535136f6dfe0678499012e853ee9abcab1./usr/local/python./usr/local/python.-m#ipykernel_launcher  (Python Path: /usr/local/bin/python, Unknown, 3.8.17) for '/work/python/sample/sample_mapping_ipynb/sample_mapping_1.ipynb' (disableUI=false)
01:14:57.286 [info] Launching server
01:14:57.288 [info] Starting Notebook
01:14:57.290 [info] Generating custom default config at /tmp/9a698a07-a9ae-42c6-bfc5-0a3940a4a174/jupyter_notebook_config.py
01:14:57.347 [warn] Failed to get activated env vars for /usr/local/bin/python in 53ms
01:14:57.360 [warn] Failed to get activated env vars for /usr/local/bin/python in 11ms
01:14:57.374 [info] Process Execution: /usr/local/bin/python -m jupyter notebook --no-browser --notebook-dir="/work/python/sample/sample_mapping_ipynb" --config=/tmp/9a698a07-a9ae-42c6-bfc5-0a3940a4a174/jupyter_notebook_config.py --NotebookApp.iopub_data_rate_limit=10000000000.0
01:14:57.871 [error] Failed to start the Jupyter Server Error: Jupyter Server crashed. Unable to connect. 
Error code from Jupyter: 1
[W 2024-01-18 01:14:57.701 ServerApp] A `_jupyter_server_extension_points` function was not found in notebook_shim. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.
[I 2024-01-18 01:14:57.702 ServerApp] jupyter_lsp | extension was successfully linked.
[I 2024-01-18 01:14:57.704 ServerApp] jupyter_server_terminals | extension was successfully linked.
[W 2024-01-18 01:14:57.704 LabApp] 'iopub_data_rate_limit' has moved from NotebookApp to ServerApp. This config will be passed to ServerApp. Be sure to update your config before our next release.
[W 2024-01-18 01:14:57.706 ServerApp] ServerApp.iopub_data_rate_limit config is deprecated in 2.0. Use ZMQChannelsWebsocketConnection.iopub_data_rate_limit.
[I 2024-01-18 01:14:57.706 ServerApp] jupyterlab | extension was successfully linked.
[I 2024-01-18 01:14:57.707 ServerApp] notebook | extension was successfully linked.
[I 2024-01-18 01:14:57.810 ServerApp] notebook_shim | extension was successfully linked.
[I 2024-01-18 01:14:57.816 ServerApp] notebook_shim | extension was successfully loaded.
[I 2024-01-18 01:14:57.817 ServerApp] jupyter_lsp | extension was successfully loaded.
[I 2024-01-18 01:14:57.817 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2024-01-18 01:14:57.818 LabApp] JupyterLab extension loaded from /usr/local/lib/python3.8/site-packages/jupyterlab
[I 2024-01-18 01:14:57.818 LabApp] JupyterLab application directory is /usr/local/share/jupyter/lab
[I 2024-01-18 01:14:57.818 LabApp] Extension Manager is 'pypi'.
[I 2024-01-18 01:14:57.820 ServerApp] jupyterlab | extension was successfully loaded.
[I 2024-01-18 01:14:57.821 ServerApp] notebook | extension was successfully loaded.
[C 2024-01-18 01:14:57.821 ServerApp] Running as <username> is not recommended. Use --allow-<username> to bypass.
    > at BD.rejectStartPromise (~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js:235:2315)
    > interpreter = [object Object]
01:14:57.871 [warn] Error occurred while trying to start the kernel, options.disableUI=false Error: Jupyter Server crashed. Unable to connect. 
Error code from Jupyter: 1
[W 2024-01-18 01:14:57.701 ServerApp] A `_jupyter_server_extension_points` function was not found in notebook_shim. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.
[I 2024-01-18 01:14:57.702 ServerApp] jupyter_lsp | extension was successfully linked.
[I 2024-01-18 01:14:57.704 ServerApp] jupyter_server_terminals | extension was successfully linked.
[W 2024-01-18 01:14:57.704 LabApp] 'iopub_data_rate_limit' has moved from NotebookApp to ServerApp. This config will be passed to ServerApp. Be sure to update your config before our next release.
[W 2024-01-18 01:14:57.706 ServerApp] ServerApp.iopub_data_rate_limit config is deprecated in 2.0. Use ZMQChannelsWebsocketConnection.iopub_data_rate_limit.
[I 2024-01-18 01:14:57.706 ServerApp] jupyterlab | extension was successfully linked.
[I 2024-01-18 01:14:57.707 ServerApp] notebook | extension was successfully linked.
[I 2024-01-18 01:14:57.810 ServerApp] notebook_shim | extension was successfully linked.
[I 2024-01-18 01:14:57.816 ServerApp] notebook_shim | extension was successfully loaded.
[I 2024-01-18 01:14:57.817 ServerApp] jupyter_lsp | extension was successfully loaded.
[I 2024-01-18 01:14:57.817 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2024-01-18 01:14:57.818 LabApp] JupyterLab extension loaded from /usr/local/lib/python3.8/site-packages/jupyterlab
[I 2024-01-18 01:14:57.818 LabApp] JupyterLab application directory is /usr/local/share/jupyter/lab
[I 2024-01-18 01:14:57.818 LabApp] Extension Manager is 'pypi'.
[I 2024-01-18 01:14:57.820 ServerApp] jupyterlab | extension was successfully loaded.
[I 2024-01-18 01:14:57.821 ServerApp] notebook | extension was successfully loaded.
[C 2024-01-18 01:14:57.821 ServerApp] Running as <username> is not recommended. Use --allow-<username> to bypass.
    > at BD.rejectStartPromise (~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js:235:2315)
    > interpreter = [object Object]
01:14:57.872 [warn] Kernel Error, context = start Error: Jupyter Server crashed. Unable to connect. 
Error code from Jupyter: 1
[W 2024-01-18 01:14:57.701 ServerApp] A `_jupyter_server_extension_points` function was not found in notebook_shim. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.
[I 2024-01-18 01:14:57.702 ServerApp] jupyter_lsp | extension was successfully linked.
[I 2024-01-18 01:14:57.704 ServerApp] jupyter_server_terminals | extension was successfully linked.
[W 2024-01-18 01:14:57.704 LabApp] 'iopub_data_rate_limit' has moved from NotebookApp to ServerApp. This config will be passed to ServerApp. Be sure to update your config before our next release.
[W 2024-01-18 01:14:57.706 ServerApp] ServerApp.iopub_data_rate_limit config is deprecated in 2.0. Use ZMQChannelsWebsocketConnection.iopub_data_rate_limit.
[I 2024-01-18 01:14:57.706 ServerApp] jupyterlab | extension was successfully linked.
[I 2024-01-18 01:14:57.707 ServerApp] notebook | extension was successfully linked.
[I 2024-01-18 01:14:57.810 ServerApp] notebook_shim | extension was successfully linked.
[I 2024-01-18 01:14:57.816 ServerApp] notebook_shim | extension was successfully loaded.
[I 2024-01-18 01:14:57.817 ServerApp] jupyter_lsp | extension was successfully loaded.
[I 2024-01-18 01:14:57.817 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2024-01-18 01:14:57.818 LabApp] JupyterLab extension loaded from /usr/local/lib/python3.8/site-packages/jupyterlab
[I 2024-01-18 01:14:57.818 LabApp] JupyterLab application directory is /usr/local/share/jupyter/lab
[I 2024-01-18 01:14:57.818 LabApp] Extension Manager is 'pypi'.
[I 2024-01-18 01:14:57.820 ServerApp] jupyterlab | extension was successfully loaded.
[I 2024-01-18 01:14:57.821 ServerApp] notebook | extension was successfully loaded.
[C 2024-01-18 01:14:57.821 ServerApp] Running as <username> is not recommended. Use --allow-<username> to bypass.
    > at BD.rejectStartPromise (~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js:235:2315)
    > interpreter = [object Object]
01:14:57.884 [info] Process Execution: /usr/local/bin/python -c "import ipykernel;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
01:14:57.981 [info] Dispose Kernel '/work/python/sample/sample_mapping_ipynb/sample_mapping_1.ipynb' associated with '/work/python/sample/sample_mapping_ipynb/sample_mapping_1.ipynb'
01:14:57.982 [error] Error in execution Error: Jupyter Server crashed. Unable to connect. 
Error code from Jupyter: 1
[W 2024-01-18 01:14:57.701 ServerApp] A `_jupyter_server_extension_points` function was not found in notebook_shim. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.
[I 2024-01-18 01:14:57.702 ServerApp] jupyter_lsp | extension was successfully linked.
[I 2024-01-18 01:14:57.704 ServerApp] jupyter_server_terminals | extension was successfully linked.
[W 2024-01-18 01:14:57.704 LabApp] 'iopub_data_rate_limit' has moved from NotebookApp to ServerApp. This config will be passed to ServerApp. Be sure to update your config before our next release.
[W 2024-01-18 01:14:57.706 ServerApp] ServerApp.iopub_data_rate_limit config is deprecated in 2.0. Use ZMQChannelsWebsocketConnection.iopub_data_rate_limit.
[I 2024-01-18 01:14:57.706 ServerApp] jupyterlab | extension was successfully linked.
[I 2024-01-18 01:14:57.707 ServerApp] notebook | extension was successfully linked.
[I 2024-01-18 01:14:57.810 ServerApp] notebook_shim | extension was successfully linked.
[I 2024-01-18 01:14:57.816 ServerApp] notebook_shim | extension was successfully loaded.
[I 2024-01-18 01:14:57.817 ServerApp] jupyter_lsp | extension was successfully loaded.
[I 2024-01-18 01:14:57.817 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2024-01-18 01:14:57.818 LabApp] JupyterLab extension loaded from /usr/local/lib/python3.8/site-packages/jupyterlab
[I 2024-01-18 01:14:57.818 LabApp] JupyterLab application directory is /usr/local/share/jupyter/lab
[I 2024-01-18 01:14:57.818 LabApp] Extension Manager is 'pypi'.
[I 2024-01-18 01:14:57.820 ServerApp] jupyterlab | extension was successfully loaded.
[I 2024-01-18 01:14:57.821 ServerApp] notebook | extension was successfully loaded.
[C 2024-01-18 01:14:57.821 ServerApp] Running as <username> is not recommended. Use --allow-<username> to bypass.
    > at BD.rejectStartPromise (~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js:235:2315)
    > interpreter = [object Object]
01:14:57.983 [error] Error in execution (get message for cell) Error: Jupyter Server crashed. Unable to connect. 
Error code from Jupyter: 1
[W 2024-01-18 01:14:57.701 ServerApp] A `_jupyter_server_extension_points` function was not found in notebook_shim. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.
[I 2024-01-18 01:14:57.702 ServerApp] jupyter_lsp | extension was successfully linked.
[I 2024-01-18 01:14:57.704 ServerApp] jupyter_server_terminals | extension was successfully linked.
[W 2024-01-18 01:14:57.704 LabApp] 'iopub_data_rate_limit' has moved from NotebookApp to ServerApp. This config will be passed to ServerApp. Be sure to update your config before our next release.
[W 2024-01-18 01:14:57.706 ServerApp] ServerApp.iopub_data_rate_limit config is deprecated in 2.0. Use ZMQChannelsWebsocketConnection.iopub_data_rate_limit.
[I 2024-01-18 01:14:57.706 ServerApp] jupyterlab | extension was successfully linked.
[I 2024-01-18 01:14:57.707 ServerApp] notebook | extension was successfully linked.
[I 2024-01-18 01:14:57.810 ServerApp] notebook_shim | extension was successfully linked.
[I 2024-01-18 01:14:57.816 ServerApp] notebook_shim | extension was successfully loaded.
[I 2024-01-18 01:14:57.817 ServerApp] jupyter_lsp | extension was successfully loaded.
[I 2024-01-18 01:14:57.817 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2024-01-18 01:14:57.818 LabApp] JupyterLab extension loaded from /usr/local/lib/python3.8/site-packages/jupyterlab
[I 2024-01-18 01:14:57.818 LabApp] JupyterLab application directory is /usr/local/share/jupyter/lab
[I 2024-01-18 01:14:57.818 LabApp] Extension Manager is 'pypi'.
[I 2024-01-18 01:14:57.820 ServerApp] jupyterlab | extension was successfully loaded.
[I 2024-01-18 01:14:57.821 ServerApp] notebook | extension was successfully loaded.
[C 2024-01-18 01:14:57.821 ServerApp] Running as <username> is not recommended. Use --allow-<username> to bypass.
    > at BD.rejectStartPromise (~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js:235:2315)
    > interpreter = [object Object]
01:14:58.006 [info] End cell 10 execution after 0s, completed @ undefined, started @ undefined
```


### Coding Language and Runtime Version

Python 3.8.17

### Language Extension Version (if applicable)

Python Extension v2023.22.1

### Anaconda Version (if applicable)

_No response_

### Running Jupyter locally or remotely?

Remote

It seems to occur only on Mac chip (M1/M2/M3).

Here is the the output from the Jupyter output panel, run in docker container on macbook pro M3max. 
```
Visual Studio Code (1.85.1, attached-container, desktop)
Jupyter Extension Version: 2023.11.1100101639.
Python Extension Version: 2023.22.1.
Pylance Extension Version: 2023.12.1.
Platform: linux (arm64).
Workspace folder /work, Home = /root
00:58:05.284 [warn] Exception while attempting zmq : /lib/aarch64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by ~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/node_modules/zeromq/prebuilds/linux-arm64/node.napi.glibc.node)
00:58:05.284 [warn] Exception while attempting zmq (fallback) : Cannot find module 'node-gyp-build'
Require stack:
- ~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/node_modules/zeromqold/lib/native.js
- ~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/node_modules/zeromqold/lib/index.js
- ~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js
- ~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.proxy.js
- ~/.vscode-server/bin/0ee08df0cf4527e40edc9aa28f4b5bd38bbff2b2/out/vs/loader.js
- ~/.vscode-server/bin/0ee08df0cf4527e40edc9aa28f4b5bd38bbff2b2/out/bootstrap-amd.js
- ~/.vscode-server/bin/0ee08df0cf4527e40edc9aa28f4b5bd38bbff2b2/out/bootstrap-fork.js
00:58:05.537 [info] Start refreshing Kernel Picker (1705539485537)
00:58:05.539 [info] Using Pylance
00:58:05.599 [warn] Failed to get activated env vars for /usr/bin/python3 in 45ms
00:58:05.608 [info] Process Execution: /usr/bin/python3 -c "import site;print("USER_BASE_VALUE");print(site.USER_BASE);print("USER_BASE_VALUE");"
00:58:05.634 [info] Process Execution: /usr/bin/python3 -c "import jupyter;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
00:58:05.638 [info] Process Execution: /usr/bin/python3 -c "import notebook;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
00:58:05.645 [info] Process Execution: /usr/bin/python3 -m pip list
00:58:05.954 [warn] No interpreter for Pylance for Notebook URI "/work/python/sample/sample_mapping_ipynb/sample_mapping_1.ipynb"
00:58:06.555 [info] End refreshing Kernel Picker (1705539485537)
00:58:08.006 [info] Start refreshing Kernel Picker (1705539488006)
00:58:10.112 [warn] Disposing old controller startUsingPythonInterpreter:'.jvsc74a57bd070a07a42c28225b490c39e102d4e2a545ef880d96770b11e09e5cb8539d9f173./usr/include/python3.7./usr/include/python3.7.-m#ipykernel_launcher' for view = 'jupyter-notebook'
00:58:10.113 [warn] Disposing old controller startUsingPythonInterpreter:'.jvsc74a57bd070a07a42c28225b490c39e102d4e2a545ef880d96770b11e09e5cb8539d9f173./usr/include/python3.7./usr/include/python3.7.-m#ipykernel_launcher (Interactive)' for view = 'interactive'
00:58:10.179 [error] Failed to get env details from Python API for /usr/include/python3.7 without an error
00:58:10.179 [error] Failed to get env details from Python API for /usr/include/python3.7 without an error
00:58:10.179 [error] Failed to get env details from Python API for /usr/include/python3.7 without an error
00:58:10.179 [error] Failed to get env details from Python API for /usr/include/python3.7 without an error
00:58:10.179 [error] Failed to get env details from Python API for /usr/include/python3.7 without an error
00:58:11.156 [info] Handle Execution of Cells 5 for /work/python/sample/sample_mapping_ipynb/sample_mapping_1.ipynb
00:58:11.188 [info] Starting Kernel startUsingPythonInterpreter, .jvsc74a57bd0949777d72b0d2535278d3dc13498b2535136f6dfe0678499012e853ee9abcab1./usr/local/python./usr/local/python.-m#ipykernel_launcher  (Python Path: /usr/local/bin/python, Unknown, 3.8.17) for '/work/python/sample/sample_mapping_ipynb/sample_mapping_1.ipynb' (disableUI=false)
00:58:11.192 [info] End refreshing Kernel Picker (1705539488006)
00:58:11.250 [warn] Failed to get activated env vars for /usr/local/bin/python in 104ms
00:58:11.260 [info] Process Execution: /usr/local/bin/python -c "import site;print("USER_BASE_VALUE");print(site.USER_BASE);print("USER_BASE_VALUE");"
00:58:11.262 [warn] Failed to get activated env vars for /usr/bin/python3 in 71ms
00:58:11.270 [info] Process Execution: /usr/bin/python3 -c "import jupyter;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
00:58:11.277 [info] Process Execution: /usr/bin/python3 -c "import notebook;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
00:58:11.287 [info] Process Execution: /usr/local/bin/python -m pip list
00:58:11.296 [warn] Error occurred while trying to start the kernel, options.disableUI=false Error: Jupyter cannot be started. Error attempting to locate Jupyter: Running cells with '/usr/bin/python3' requires the jupyter and notebook package.
    > at Zv.startServerImpl (~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js:234:10443)
00:58:11.296 [warn] Kernel Error, context = start Error: Jupyter cannot be started. Error attempting to locate Jupyter: Running cells with '/usr/bin/python3' requires the jupyter and notebook package.
    > at Zv.startServerImpl (~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js:234:10443)
00:58:11.308 [info] Process Execution: /usr/bin/python3 -c "import pip;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
00:58:11.315 [info] Process Execution: /usr/bin/python3 -c "import jupyter;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
00:58:11.321 [info] Process Execution: /usr/bin/python3 -c "import notebook;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
00:58:18.007 [info] Dispose Kernel '/work/python/sample/sample_mapping_ipynb/sample_mapping_1.ipynb' associated with '/work/python/sample/sample_mapping_ipynb/sample_mapping_1.ipynb'
00:58:18.009 [error] Error in execution Error: Jupyter cannot be started. Error attempting to locate Jupyter: Running cells with '/usr/bin/python3' requires the jupyter and notebook package.
    > at Zv.startServerImpl (~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js:234:10443)
00:58:18.010 [error] Error in execution (get message for cell) Error: Jupyter cannot be started. Error attempting to locate Jupyter: Running cells with '/usr/bin/python3' requires the jupyter and notebook package.
    > at Zv.startServerImpl (~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js:234:10443)
00:58:18.049 [info] End cell 5 execution after 0s, completed @ undefined, started @ undefined
00:58:31.403 [info] Handle Execution of Cells 5 for /work/python/sample/sample_mapping_ipynb/sample_mapping_1.ipynb
00:58:31.407 [info] Starting Kernel startUsingPythonInterpreter, .jvsc74a57bd0949777d72b0d2535278d3dc13498b2535136f6dfe0678499012e853ee9abcab1./usr/local/python./usr/local/python.-m#ipykernel_launcher  (Python Path: /usr/local/bin/python, Unknown, 3.8.17) for '/work/python/sample/sample_mapping_ipynb/sample_mapping_1.ipynb' (disableUI=false)
00:58:31.435 [info] Process Execution: /usr/bin/python3 -c "import jupyter;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
00:58:31.443 [info] Process Execution: /usr/bin/python3 -c "import notebook;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
00:58:31.455 [warn] Error occurred while trying to start the kernel, options.disableUI=false Error: Jupyter cannot be started. Error attempting to locate Jupyter: Running cells with '/usr/bin/python3' requires the jupyter and notebook package.
    > at Zv.startServerImpl (~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js:234:10443)
00:58:31.457 [warn] Kernel Error, context = start Error: Jupyter cannot be started. Error attempting to locate Jupyter: Running cells with '/usr/bin/python3' requires the jupyter and notebook package.
    > at Zv.startServerImpl (~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js:234:10443)
00:58:31.471 [info] Process Execution: /usr/bin/python3 -c "import pip;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
00:58:31.477 [info] Process Execution: /usr/bin/python3 -c "import jupyter;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
00:58:31.482 [info] Process Execution: /usr/bin/python3 -c "import notebook;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
00:58:36.820 [info] Process Execution: /usr/local/bin/python -c "import pip;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
00:58:36.828 [info] Process Execution: /usr/local/bin/python -c "import jupyter;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
00:58:36.835 [info] Process Execution: /usr/local/bin/python -c "import notebook;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
00:58:39.028 [info] Process Execution: /usr/local/bin/python -c "import pip;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
00:58:39.066 [info] Process Execution: /usr/local/bin/python -m pip install -U --user notebook jupyter-client<8 pyzmq<25
00:58:48.186 [info] Process Execution: /usr/local/bin/python -c "import notebook;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
00:58:48.208 [info] Process Execution: /usr/local/bin/python -m pip list
00:58:48.213 [info] Process Execution: /usr/local/bin/python -m jupyter kernelspec --version
00:58:48.325 [info] Dispose Kernel '/work/python/sample/sample_mapping_ipynb/sample_mapping_1.ipynb' associated with '/work/python/sample/sample_mapping_ipynb/sample_mapping_1.ipynb'
00:58:48.327 [info] Starting Kernel startUsingPythonInterpreter, .jvsc74a57bd0949777d72b0d2535278d3dc13498b2535136f6dfe0678499012e853ee9abcab1./usr/local/python./usr/local/python.-m#ipykernel_launcher  (Python Path: /usr/local/bin/python, Unknown, 3.8.17) for '/work/python/sample/sample_mapping_ipynb/sample_mapping_1.ipynb' (disableUI=false)
00:58:48.328 [info] Launching server
00:58:48.330 [info] Starting Notebook
00:58:48.331 [info] Generating custom default config at /tmp/7f3ae3e6-7829-4c5b-a11d-490fc1bbe002/jupyter_notebook_config.py
00:58:48.381 [warn] Failed to get activated env vars for /usr/local/bin/python in 55ms
00:58:48.390 [info] Process Execution: /usr/local/bin/python -m jupyter notebook --no-browser --notebook-dir="/work/python/sample/sample_mapping_ipynb" --config=/tmp/7f3ae3e6-7829-4c5b-a11d-490fc1bbe002/jupyter_notebook_config.py --NotebookApp.iopub_data_rate_limit=10000000000.0
00:58:48.832 [error] Failed to start the Jupyter Server Error: Jupyter Server crashed. Unable to connect. 
Error code from Jupyter: 1
[W 2024-01-18 00:58:48.654 ServerApp] A `_jupyter_server_extension_points` function was not found in notebook_shim. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.
[I 2024-01-18 00:58:48.654 ServerApp] jupyter_lsp | extension was successfully linked.
[I 2024-01-18 00:58:48.656 ServerApp] jupyter_server_terminals | extension was successfully linked.
[W 2024-01-18 00:58:48.656 LabApp] 'iopub_data_rate_limit' has moved from NotebookApp to ServerApp. This config will be passed to ServerApp. Be sure to update your config before our next release.
[W 2024-01-18 00:58:48.658 ServerApp] ServerApp.iopub_data_rate_limit config is deprecated in 2.0. Use ZMQChannelsWebsocketConnection.iopub_data_rate_limit.
[I 2024-01-18 00:58:48.658 ServerApp] jupyterlab | extension was successfully linked.
[I 2024-01-18 00:58:48.659 ServerApp] notebook | extension was successfully linked.
[I 2024-01-18 00:58:48.660 ServerApp] Writing Jupyter server cookie secret to ~/.local/share/jupyter/runtime/jupyter_cookie_secret
[I 2024-01-18 00:58:48.760 ServerApp] notebook_shim | extension was successfully linked.
[I 2024-01-18 00:58:48.766 ServerApp] notebook_shim | extension was successfully loaded.
[I 2024-01-18 00:58:48.767 ServerApp] jupyter_lsp | extension was successfully loaded.
[I 2024-01-18 00:58:48.767 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2024-01-18 00:58:48.768 LabApp] JupyterLab extension loaded from ~/.local/lib/python3.8/site-packages/jupyterlab
[I 2024-01-18 00:58:48.768 LabApp] JupyterLab application directory is ~/.local/share/jupyter/lab
[I 2024-01-18 00:58:48.768 LabApp] Extension Manager is 'pypi'.
[I 2024-01-18 00:58:48.781 ServerApp] jupyterlab | extension was successfully loaded.
[I 2024-01-18 00:58:48.782 ServerApp] notebook | extension was successfully loaded.
[C 2024-01-18 00:58:48.782 ServerApp] Running as <username> is not recommended. Use --allow-<username> to bypass.
    > at BD.rejectStartPromise (~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js:235:2315)
    > interpreter = [object Object]
00:58:48.833 [warn] Error occurred while trying to start the kernel, options.disableUI=false Error: Jupyter Server crashed. Unable to connect. 
Error code from Jupyter: 1
[W 2024-01-18 00:58:48.654 ServerApp] A `_jupyter_server_extension_points` function was not found in notebook_shim. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.
[I 2024-01-18 00:58:48.654 ServerApp] jupyter_lsp | extension was successfully linked.
[I 2024-01-18 00:58:48.656 ServerApp] jupyter_server_terminals | extension was successfully linked.
[W 2024-01-18 00:58:48.656 LabApp] 'iopub_data_rate_limit' has moved from NotebookApp to ServerApp. This config will be passed to ServerApp. Be sure to update your config before our next release.
[W 2024-01-18 00:58:48.658 ServerApp] ServerApp.iopub_data_rate_limit config is deprecated in 2.0. Use ZMQChannelsWebsocketConnection.iopub_data_rate_limit.
[I 2024-01-18 00:58:48.658 ServerApp] jupyterlab | extension was successfully linked.
[I 2024-01-18 00:58:48.659 ServerApp] notebook | extension was successfully linked.
[I 2024-01-18 00:58:48.660 ServerApp] Writing Jupyter server cookie secret to ~/.local/share/jupyter/runtime/jupyter_cookie_secret
[I 2024-01-18 00:58:48.760 ServerApp] notebook_shim | extension was successfully linked.
[I 2024-01-18 00:58:48.766 ServerApp] notebook_shim | extension was successfully loaded.
[I 2024-01-18 00:58:48.767 ServerApp] jupyter_lsp | extension was successfully loaded.
[I 2024-01-18 00:58:48.767 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2024-01-18 00:58:48.768 LabApp] JupyterLab extension loaded from ~/.local/lib/python3.8/site-packages/jupyterlab
[I 2024-01-18 00:58:48.768 LabApp] JupyterLab application directory is ~/.local/share/jupyter/lab
[I 2024-01-18 00:58:48.768 LabApp] Extension Manager is 'pypi'.
[I 2024-01-18 00:58:48.781 ServerApp] jupyterlab | extension was successfully loaded.
[I 2024-01-18 00:58:48.782 ServerApp] notebook | extension was successfully loaded.
[C 2024-01-18 00:58:48.782 ServerApp] Running as <username> is not recommended. Use --allow-<username> to bypass.
    > at BD.rejectStartPromise (~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js:235:2315)
    > interpreter = [object Object]
00:58:48.833 [warn] Kernel Error, context = start Error: Jupyter Server crashed. Unable to connect. 
Error code from Jupyter: 1
[W 2024-01-18 00:58:48.654 ServerApp] A `_jupyter_server_extension_points` function was not found in notebook_shim. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.
[I 2024-01-18 00:58:48.654 ServerApp] jupyter_lsp | extension was successfully linked.
[I 2024-01-18 00:58:48.656 ServerApp] jupyter_server_terminals | extension was successfully linked.
[W 2024-01-18 00:58:48.656 LabApp] 'iopub_data_rate_limit' has moved from NotebookApp to ServerApp. This config will be passed to ServerApp. Be sure to update your config before our next release.
[W 2024-01-18 00:58:48.658 ServerApp] ServerApp.iopub_data_rate_limit config is deprecated in 2.0. Use ZMQChannelsWebsocketConnection.iopub_data_rate_limit.
[I 2024-01-18 00:58:48.658 ServerApp] jupyterlab | extension was successfully linked.
[I 2024-01-18 00:58:48.659 ServerApp] notebook | extension was successfully linked.
[I 2024-01-18 00:58:48.660 ServerApp] Writing Jupyter server cookie secret to ~/.local/share/jupyter/runtime/jupyter_cookie_secret
[I 2024-01-18 00:58:48.760 ServerApp] notebook_shim | extension was successfully linked.
[I 2024-01-18 00:58:48.766 ServerApp] notebook_shim | extension was successfully loaded.
[I 2024-01-18 00:58:48.767 ServerApp] jupyter_lsp | extension was successfully loaded.
[I 2024-01-18 00:58:48.767 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2024-01-18 00:58:48.768 LabApp] JupyterLab extension loaded from ~/.local/lib/python3.8/site-packages/jupyterlab
[I 2024-01-18 00:58:48.768 LabApp] JupyterLab application directory is ~/.local/share/jupyter/lab
[I 2024-01-18 00:58:48.768 LabApp] Extension Manager is 'pypi'.
[I 2024-01-18 00:58:48.781 ServerApp] jupyterlab | extension was successfully loaded.
[I 2024-01-18 00:58:48.782 ServerApp] notebook | extension was successfully loaded.
[C 2024-01-18 00:58:48.782 ServerApp] Running as <username> is not recommended. Use --allow-<username> to bypass.
    > at BD.rejectStartPromise (~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js:235:2315)
    > interpreter = [object Object]
00:58:48.840 [info] Process Execution: /usr/local/bin/python -c "import ipykernel;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
00:58:48.912 [info] Dispose Kernel '/work/python/sample/sample_mapping_ipynb/sample_mapping_1.ipynb' associated with '/work/python/sample/sample_mapping_ipynb/sample_mapping_1.ipynb'
00:58:48.913 [error] Error in execution Error: Jupyter Server crashed. Unable to connect. 
Error code from Jupyter: 1
[W 2024-01-18 00:58:48.654 ServerApp] A `_jupyter_server_extension_points` function was not found in notebook_shim. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.
[I 2024-01-18 00:58:48.654 ServerApp] jupyter_lsp | extension was successfully linked.
[I 2024-01-18 00:58:48.656 ServerApp] jupyter_server_terminals | extension was successfully linked.
[W 2024-01-18 00:58:48.656 LabApp] 'iopub_data_rate_limit' has moved from NotebookApp to ServerApp. This config will be passed to ServerApp. Be sure to update your config before our next release.
[W 2024-01-18 00:58:48.658 ServerApp] ServerApp.iopub_data_rate_limit config is deprecated in 2.0. Use ZMQChannelsWebsocketConnection.iopub_data_rate_limit.
[I 2024-01-18 00:58:48.658 ServerApp] jupyterlab | extension was successfully linked.
[I 2024-01-18 00:58:48.659 ServerApp] notebook | extension was successfully linked.
[I 2024-01-18 00:58:48.660 ServerApp] Writing Jupyter server cookie secret to ~/.local/share/jupyter/runtime/jupyter_cookie_secret
[I 2024-01-18 00:58:48.760 ServerApp] notebook_shim | extension was successfully linked.
[I 2024-01-18 00:58:48.766 ServerApp] notebook_shim | extension was successfully loaded.
[I 2024-01-18 00:58:48.767 ServerApp] jupyter_lsp | extension was successfully loaded.
[I 2024-01-18 00:58:48.767 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2024-01-18 00:58:48.768 LabApp] JupyterLab extension loaded from ~/.local/lib/python3.8/site-packages/jupyterlab
[I 2024-01-18 00:58:48.768 LabApp] JupyterLab application directory is ~/.local/share/jupyter/lab
[I 2024-01-18 00:58:48.768 LabApp] Extension Manager is 'pypi'.
[I 2024-01-18 00:58:48.781 ServerApp] jupyterlab | extension was successfully loaded.
[I 2024-01-18 00:58:48.782 ServerApp] notebook | extension was successfully loaded.
[C 2024-01-18 00:58:48.782 ServerApp] Running as <username> is not recommended. Use --allow-<username> to bypass.
    > at BD.rejectStartPromise (~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js:235:2315)
    > interpreter = [object Object]
00:58:48.913 [error] Error in execution (get message for cell) Error: Jupyter Server crashed. Unable to connect. 
Error code from Jupyter: 1
[W 2024-01-18 00:58:48.654 ServerApp] A `_jupyter_server_extension_points` function was not found in notebook_shim. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.
[I 2024-01-18 00:58:48.654 ServerApp] jupyter_lsp | extension was successfully linked.
[I 2024-01-18 00:58:48.656 ServerApp] jupyter_server_terminals | extension was successfully linked.
[W 2024-01-18 00:58:48.656 LabApp] 'iopub_data_rate_limit' has moved from NotebookApp to ServerApp. This config will be passed to ServerApp. Be sure to update your config before our next release.
[W 2024-01-18 00:58:48.658 ServerApp] ServerApp.iopub_data_rate_limit config is deprecated in 2.0. Use ZMQChannelsWebsocketConnection.iopub_data_rate_limit.
[I 2024-01-18 00:58:48.658 ServerApp] jupyterlab | extension was successfully linked.
[I 2024-01-18 00:58:48.659 ServerApp] notebook | extension was successfully linked.
[I 2024-01-18 00:58:48.660 ServerApp] Writing Jupyter server cookie secret to ~/.local/share/jupyter/runtime/jupyter_cookie_secret
[I 2024-01-18 00:58:48.760 ServerApp] notebook_shim | extension was successfully linked.
[I 2024-01-18 00:58:48.766 ServerApp] notebook_shim | extension was successfully loaded.
[I 2024-01-18 00:58:48.767 ServerApp] jupyter_lsp | extension was successfully loaded.
[I 2024-01-18 00:58:48.767 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2024-01-18 00:58:48.768 LabApp] JupyterLab extension loaded from ~/.local/lib/python3.8/site-packages/jupyterlab
[I 2024-01-18 00:58:48.768 LabApp] JupyterLab application directory is ~/.local/share/jupyter/lab
[I 2024-01-18 00:58:48.768 LabApp] Extension Manager is 'pypi'.
[I 2024-01-18 00:58:48.781 ServerApp] jupyterlab | extension was successfully loaded.
[I 2024-01-18 00:58:48.782 ServerApp] notebook | extension was successfully loaded.
[C 2024-01-18 00:58:48.782 ServerApp] Running as <username> is not recommended. Use --allow-<username> to bypass.
    > at BD.rejectStartPromise (~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js:235:2315)
    > interpreter = [object Object]
00:58:48.936 [info] End cell 5 execution after 0s, completed @ undefined, started @ undefined
00:58:51.434 [info] Handle Execution of Cells 5 for /work/python/sample/sample_mapping_ipynb/sample_mapping_1.ipynb
00:58:51.437 [info] Starting Kernel startUsingPythonInterpreter, .jvsc74a57bd0949777d72b0d2535278d3dc13498b2535136f6dfe0678499012e853ee9abcab1./usr/local/python./usr/local/python.-m#ipykernel_launcher  (Python Path: /usr/local/bin/python, Unknown, 3.8.17) for '/work/python/sample/sample_mapping_ipynb/sample_mapping_1.ipynb' (disableUI=false)
00:58:51.437 [info] Launching server
00:58:51.439 [info] Starting Notebook
00:58:51.440 [info] Generating custom default config at /tmp/6388282f-3308-4652-a60b-26fb23d8468c/jupyter_notebook_config.py
00:58:51.457 [info] Process Execution: /usr/local/bin/python -m jupyter notebook --no-browser --notebook-dir="/work/python/sample/sample_mapping_ipynb" --config=/tmp/6388282f-3308-4652-a60b-26fb23d8468c/jupyter_notebook_config.py --NotebookApp.iopub_data_rate_limit=10000000000.0
00:58:51.838 [error] Failed to start the Jupyter Server Error: Jupyter Server crashed. Unable to connect. 
Error code from Jupyter: 1
[W 2024-01-18 00:58:51.718 ServerApp] A `_jupyter_server_extension_points` function was not found in notebook_shim. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.
[I 2024-01-18 00:58:51.719 ServerApp] jupyter_lsp | extension was successfully linked.
[I 2024-01-18 00:58:51.720 ServerApp] jupyter_server_terminals | extension was successfully linked.
[W 2024-01-18 00:58:51.721 LabApp] 'iopub_data_rate_limit' has moved from NotebookApp to ServerApp. This config will be passed to ServerApp. Be sure to update your config before our next release.
[W 2024-01-18 00:58:51.722 ServerApp] ServerApp.iopub_data_rate_limit config is deprecated in 2.0. Use ZMQChannelsWebsocketConnection.iopub_data_rate_limit.
[I 2024-01-18 00:58:51.722 ServerApp] jupyterlab | extension was successfully linked.
[I 2024-01-18 00:58:51.724 ServerApp] notebook | extension was successfully linked.
[I 2024-01-18 00:58:51.787 ServerApp] notebook_shim | extension was successfully linked.
[I 2024-01-18 00:58:51.792 ServerApp] notebook_shim | extension was successfully loaded.
[I 2024-01-18 00:58:51.793 ServerApp] jupyter_lsp | extension was successfully loaded.
[I 2024-01-18 00:58:51.794 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2024-01-18 00:58:51.794 LabApp] JupyterLab extension loaded from ~/.local/lib/python3.8/site-packages/jupyterlab
[I 2024-01-18 00:58:51.794 LabApp] JupyterLab application directory is ~/.local/share/jupyter/lab
[I 2024-01-18 00:58:51.794 LabApp] Extension Manager is 'pypi'.
[I 2024-01-18 00:58:51.795 ServerApp] jupyterlab | extension was successfully loaded.
[I 2024-01-18 00:58:51.796 ServerApp] notebook | extension was successfully loaded.
[C 2024-01-18 00:58:51.796 ServerApp] Running as <username> is not recommended. Use --allow-<username> to bypass.
    > at BD.rejectStartPromise (~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js:235:2315)
    > interpreter = [object Object]
00:58:51.839 [warn] Error occurred while trying to start the kernel, options.disableUI=false Error: Jupyter Server crashed. Unable to connect. 
Error code from Jupyter: 1
[W 2024-01-18 00:58:51.718 ServerApp] A `_jupyter_server_extension_points` function was not found in notebook_shim. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.
[I 2024-01-18 00:58:51.719 ServerApp] jupyter_lsp | extension was successfully linked.
[I 2024-01-18 00:58:51.720 ServerApp] jupyter_server_terminals | extension was successfully linked.
[W 2024-01-18 00:58:51.721 LabApp] 'iopub_data_rate_limit' has moved from NotebookApp to ServerApp. This config will be passed to ServerApp. Be sure to update your config before our next release.
[W 2024-01-18 00:58:51.722 ServerApp] ServerApp.iopub_data_rate_limit config is deprecated in 2.0. Use ZMQChannelsWebsocketConnection.iopub_data_rate_limit.
[I 2024-01-18 00:58:51.722 ServerApp] jupyterlab | extension was successfully linked.
[I 2024-01-18 00:58:51.724 ServerApp] notebook | extension was successfully linked.
[I 2024-01-18 00:58:51.787 ServerApp] notebook_shim | extension was successfully linked.
[I 2024-01-18 00:58:51.792 ServerApp] notebook_shim | extension was successfully loaded.
[I 2024-01-18 00:58:51.793 ServerApp] jupyter_lsp | extension was successfully loaded.
[I 2024-01-18 00:58:51.794 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2024-01-18 00:58:51.794 LabApp] JupyterLab extension loaded from ~/.local/lib/python3.8/site-packages/jupyterlab
[I 2024-01-18 00:58:51.794 LabApp] JupyterLab application directory is ~/.local/share/jupyter/lab
[I 2024-01-18 00:58:51.794 LabApp] Extension Manager is 'pypi'.
[I 2024-01-18 00:58:51.795 ServerApp] jupyterlab | extension was successfully loaded.
[I 2024-01-18 00:58:51.796 ServerApp] notebook | extension was successfully loaded.
[C 2024-01-18 00:58:51.796 ServerApp] Running as <username> is not recommended. Use --allow-<username> to bypass.
    > at BD.rejectStartPromise (~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js:235:2315)
    > interpreter = [object Object]
00:58:51.839 [warn] Kernel Error, context = start Error: Jupyter Server crashed. Unable to connect. 
Error code from Jupyter: 1
[W 2024-01-18 00:58:51.718 ServerApp] A `_jupyter_server_extension_points` function was not found in notebook_shim. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.
[I 2024-01-18 00:58:51.719 ServerApp] jupyter_lsp | extension was successfully linked.
[I 2024-01-18 00:58:51.720 ServerApp] jupyter_server_terminals | extension was successfully linked.
[W 2024-01-18 00:58:51.721 LabApp] 'iopub_data_rate_limit' has moved from NotebookApp to ServerApp. This config will be passed to ServerApp. Be sure to update your config before our next release.
[W 2024-01-18 00:58:51.722 ServerApp] ServerApp.iopub_data_rate_limit config is deprecated in 2.0. Use ZMQChannelsWebsocketConnection.iopub_data_rate_limit.
[I 2024-01-18 00:58:51.722 ServerApp] jupyterlab | extension was successfully linked.
[I 2024-01-18 00:58:51.724 ServerApp] notebook | extension was successfully linked.
[I 2024-01-18 00:58:51.787 ServerApp] notebook_shim | extension was successfully linked.
[I 2024-01-18 00:58:51.792 ServerApp] notebook_shim | extension was successfully loaded.
[I 2024-01-18 00:58:51.793 ServerApp] jupyter_lsp | extension was successfully loaded.
[I 2024-01-18 00:58:51.794 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2024-01-18 00:58:51.794 LabApp] JupyterLab extension loaded from ~/.local/lib/python3.8/site-packages/jupyterlab
[I 2024-01-18 00:58:51.794 LabApp] JupyterLab application directory is ~/.local/share/jupyter/lab
[I 2024-01-18 00:58:51.794 LabApp] Extension Manager is 'pypi'.
[I 2024-01-18 00:58:51.795 ServerApp] jupyterlab | extension was successfully loaded.
[I 2024-01-18 00:58:51.796 ServerApp] notebook | extension was successfully loaded.
[C 2024-01-18 00:58:51.796 ServerApp] Running as <username> is not recommended. Use --allow-<username> to bypass.
    > at BD.rejectStartPromise (~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js:235:2315)
    > interpreter = [object Object]
00:58:51.844 [info] Process Execution: /usr/local/bin/python -c "import ipykernel;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
00:58:51.913 [info] Dispose Kernel '/work/python/sample/sample_mapping_ipynb/sample_mapping_1.ipynb' associated with '/work/python/sample/sample_mapping_ipynb/sample_mapping_1.ipynb'
00:58:51.913 [error] Error in execution Error: Jupyter Server crashed. Unable to connect. 
Error code from Jupyter: 1
[W 2024-01-18 00:58:51.718 ServerApp] A `_jupyter_server_extension_points` function was not found in notebook_shim. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.
[I 2024-01-18 00:58:51.719 ServerApp] jupyter_lsp | extension was successfully linked.
[I 2024-01-18 00:58:51.720 ServerApp] jupyter_server_terminals | extension was successfully linked.
[W 2024-01-18 00:58:51.721 LabApp] 'iopub_data_rate_limit' has moved from NotebookApp to ServerApp. This config will be passed to ServerApp. Be sure to update your config before our next release.
[W 2024-01-18 00:58:51.722 ServerApp] ServerApp.iopub_data_rate_limit config is deprecated in 2.0. Use ZMQChannelsWebsocketConnection.iopub_data_rate_limit.
[I 2024-01-18 00:58:51.722 ServerApp] jupyterlab | extension was successfully linked.
[I 2024-01-18 00:58:51.724 ServerApp] notebook | extension was successfully linked.
[I 2024-01-18 00:58:51.787 ServerApp] notebook_shim | extension was successfully linked.
[I 2024-01-18 00:58:51.792 ServerApp] notebook_shim | extension was successfully loaded.
[I 2024-01-18 00:58:51.793 ServerApp] jupyter_lsp | extension was successfully loaded.
[I 2024-01-18 00:58:51.794 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2024-01-18 00:58:51.794 LabApp] JupyterLab extension loaded from ~/.local/lib/python3.8/site-packages/jupyterlab
[I 2024-01-18 00:58:51.794 LabApp] JupyterLab application directory is ~/.local/share/jupyter/lab
[I 2024-01-18 00:58:51.794 LabApp] Extension Manager is 'pypi'.
[I 2024-01-18 00:58:51.795 ServerApp] jupyterlab | extension was successfully loaded.
[I 2024-01-18 00:58:51.796 ServerApp] notebook | extension was successfully loaded.
[C 2024-01-18 00:58:51.796 ServerApp] Running as <username> is not recommended. Use --allow-<username> to bypass.
    > at BD.rejectStartPromise (~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js:235:2315)
    > interpreter = [object Object]
00:58:51.914 [error] Error in execution (get message for cell) Error: Jupyter Server crashed. Unable to connect. 
Error code from Jupyter: 1
[W 2024-01-18 00:58:51.718 ServerApp] A `_jupyter_server_extension_points` function was not found in notebook_shim. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.
[I 2024-01-18 00:58:51.719 ServerApp] jupyter_lsp | extension was successfully linked.
[I 2024-01-18 00:58:51.720 ServerApp] jupyter_server_terminals | extension was successfully linked.
[W 2024-01-18 00:58:51.721 LabApp] 'iopub_data_rate_limit' has moved from NotebookApp to ServerApp. This config will be passed to ServerApp. Be sure to update your config before our next release.
[W 2024-01-18 00:58:51.722 ServerApp] ServerApp.iopub_data_rate_limit config is deprecated in 2.0. Use ZMQChannelsWebsocketConnection.iopub_data_rate_limit.
[I 2024-01-18 00:58:51.722 ServerApp] jupyterlab | extension was successfully linked.
[I 2024-01-18 00:58:51.724 ServerApp] notebook | extension was successfully linked.
[I 2024-01-18 00:58:51.787 ServerApp] notebook_shim | extension was successfully linked.
[I 2024-01-18 00:58:51.792 ServerApp] notebook_shim | extension was successfully loaded.
[I 2024-01-18 00:58:51.793 ServerApp] jupyter_lsp | extension was successfully loaded.
[I 2024-01-18 00:58:51.794 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2024-01-18 00:58:51.794 LabApp] JupyterLab extension loaded from ~/.local/lib/python3.8/site-packages/jupyterlab
[I 2024-01-18 00:58:51.794 LabApp] JupyterLab application directory is ~/.local/share/jupyter/lab
[I 2024-01-18 00:58:51.794 LabApp] Extension Manager is 'pypi'.
[I 2024-01-18 00:58:51.795 ServerApp] jupyterlab | extension was successfully loaded.
[I 2024-01-18 00:58:51.796 ServerApp] notebook | extension was successfully loaded.
[C 2024-01-18 00:58:51.796 ServerApp] Running as <username> is not recommended. Use --allow-<username> to bypass.
    > at BD.rejectStartPromise (~/.vscode-server/extensions/ms-toolsai.jupyter-2023.11.1100101639-linux-arm64/dist/extension.node.js:235:2315)
    > interpreter = [object Object]
00:58:51.936 [info] End cell 5 execution after 0s, completed @ undefined, started @ undefined
03:43:24.204 [info] Start refreshing Interpreter Kernel Picker
03:43:24.205 [info] Start refreshing Kernel Picker (1705549404205)
03:43:25.224 [info] End refreshing Kernel Picker (1705549404205)

```

The Dockerfile is here
```
FROM python:3.8-slim-buster
USER root

RUN apt-get update && \
    apt-get -y install \
            python3-distutils \
            python3-dev \
            python-pip \
            python3-pip \
            gcc \
            g++ \
            curl \
            bash \
            build-essential \
            libssl-dev \
            libffi-dev \
            libproj-dev \
            proj-data \
            proj-bin \
            libgeos-dev \
            libgl1-mesa-dev \
            gdal-bin \
            python3-gdal \
            libgdal-dev

RUN pip3 install --upgrade pip
RUN pip3 install --upgrade setuptools


COPY ./python/requirements.txt .
RUN pip3 install -r requirements.txt

WORKDIR /work

```

And the requirements.txt 
```
fiona==1.8.20
geopandas==0.10.2
pygeos==0.12.0
Shapely==1.8.2
SQLAlchemy==1.4.36
psycopg2==2.9.3
matplotlib==3.5.2
mapclassify==2.4.3
folium == 0.14.0
ipykernel == 6.5.0
```

However my colleague found a solution to use an old Jupyter Extension Version (v2023.10.1100000000) and older!

--- a/build/esbuild/build.ts
+++ b/build/esbuild/build.ts
@@ -336,7 +336,14 @@ async function buildAll() {
                     });
                 })
         );
-        builders.push(copyJQuery(), copyAminya(), copyZeroMQ(), copyZeroMQOld(), buildVSCodeJsonRPC());
+        builders.push(
+            copyJQuery(),
+            copyAminya(),
+            copyZeroMQ(),
+            copyZeroMQOld(),
+            copyNodeGypBuild(),
+            buildVSCodeJsonRPC()
+        );
     }
 
     await Promise.all(builders);
@@ -383,6 +390,13 @@ async function copyZeroMQOld() {
         recursive: true,
         filter: (src) => shouldCopyFileFromZmqFolder(src)
     });
+}
+async function copyNodeGypBuild() {
+    const source = path.join(extensionFolder, 'node_modules', 'node-gyp-build');
+    const target = path.join(extensionFolder, 'dist', 'node_modules', 'node-gyp-build');
+    await fs.ensureDir(path.dirname(target));
+    await fs.ensureDir(target);
+    await fs.copy(source, target, { recursive: true });
 }
 async function buildVSCodeJsonRPC() {
     const source = path.join(extensionFolder, 'node_modules', 'vscode-jsonrpc');

